### PR TITLE
Allow setting the machine name

### PR DESCRIPTION
**Pull request by BronsonQuick**
This starts work on #628. It's gonna need some refactoring and also adding some support for Windows to but it's a bit of a start in case someone else wants to pick it up in the meantime.

### CODE REVIEW

File: puppet/chassis.rb [hunk 1]
```diff
@@ -134,6 +134,20 @@ def self.normalize_config(config)
...
 				end
 			end
+
+			machine_name = '' unless config["machine_name"]
+
+			# If we don't have a machine name in the config generate one from the directory structure.
+			if machine_name == ''
+				pwd = Dir.pwd.downcase
+				name = pwd.gsub("/","_")
+				if name[0] == "/"
```
Comment (roborourke) [inline puppet/chassis.rb]: Is this a weird side effect of how `gsub` works? Pretty annoying! Thanks for making a start on this btw

Comment (BronsonQuick) [reply puppet/chassis.rb]: Nah not at all. I was just following your comment:
>/Users/bob/chassis becomes users_bob_chassis
So I downcased it and needed to remove the first `/`

File: puppet/chassis.rb [hunk 2]
```diff
@@ -134,6 +134,15 @@ def self.normalize_config(config)
 					config["mapped_paths"][path] = config["mapped_paths"]["base"]
 				end
 			end
+
+			machine_name = '' unless config["machine_name"]
+
+			# If we don't have a machine name in the config generate one from the directory structure.
+			if machine_name == ''
+				pwd = Dir.pwd.downcase
+				# Replace all non word characters with underscores then trim underscores.
```
Comment (BronsonQuick) [inline puppet/chassis.rb]: FYI then machine name can actually contain spaces i.e. `rob is a champion` is a valid name.

File: config.yaml [hunk 3]
```diff
@@ -110,6 +110,11 @@ php: 7.3
 # Maximum file upload size. This will set post_max_size and upload_max_filesize in PHP and client_max_body_size in Nginx.
 upload_size: 1024M
 
+# Vagrant machine name
+# This can be any string and helps to identify a specific VM
+# NOTE: If changed the machine will need to be re-created and should ideally be destroyed before any change
+machine_name: default
```
Comment (rmccue) [inline config.yaml]: Will this make the machine name statically `default`? The actual default in Vagrant is `basename( directory ) + machine_name + timestamp + some_other_number`

Comment (roborourke) [reply config.yaml]: Let me double check real quick

Comment (roborourke) [reply config.yaml]: I thought it was a special value - added if statements to only apply the setting if it's not `default`

Comment (rmccue) [reply config.yaml]: 😗 👌

**Pull request merged**